api/src/routes/retrieve.js

```diff
@@ -5,6 +5,7 @@ import asyncRoute from '../utils/async-route.js';
 export default () => asyncRoute(async (req, res) => {
   const { query } = req;
   const { tenant, realm } = query;
+  const granularity = query.granularity ? query.granularity : 'week';
```

Comment:
I would also validate that the granularity is supported: 

Add the support set before the file export...
```js
const granularities = new Set(['week'; 'month']);
```

Then check inside the route:
```js
if (!granularities.has(granularity)) throw createError(400, 'The provided granularity is not supported');
```